src/generators/utils/parse.ts

```diff
@@ -65,33 +65,62 @@ function getDomain(url: string) {
     }
 }
 
+const radix = " !#$%&'()*+,-./0123456789:;=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+
+function encodeNumber(n: number, length: number | null) {
+    let s = '';
+    for (let c = n; c; c = Math.floor(c / radix.length)) {
+        s = radix[c % radix.length] + s;
+    }
+    if (length !== null) {
+        s = radix[0].repeat(length - s.length) + s;
+    }
+    return s;
+}
+
+function decodeNumber(str: string) {
+    let n = 0;
+    for (let i = 0; i < str.length; i++) {
+        n = radix.length * n + radix.indexOf(str[i]);
+    }
+    return n;
+}
+
 /*
- * Encode all offsets into a string, where each record is 7 bytes long:
- *  - 4 bytes for start offset
- *  - 3 bytes for record length (end offset - start offset)
- * Both values are stored in base 36 (radix 36) notation.
- * Maximum supported numbers:
- *  - start offset must be no more than parseInt('zzzz', 36) = 1679615
- *  - length must be no more than parseInt('zzz', 36) = 46655
- *
+ * Encode all offsets into a string, with a fixed-length header and records of equal size:
+ *  - header:
+ *    - 1 char for length of record offset (offsetLength)
+ *    - 1 char for length of record length (lengthLength)
```

Comment:
So with the current implementation they can't be larger than 92? As it would otherwise take multiple characters of space.

Reply:
This means encoded length of record offset and encoded length of record length can not be larger than 92. This means offsets and lengths themselves can not be larger than 92<sup>92</sup> - 1. That should be plenty, right?

Reply:
Yeah that should be plenty.